src/xcode/ENA/ENA/Source/Models/Exposure/__tests__/ExposureManagerTests.swift

```diff
@@ -0,0 +1,112 @@
+//
+// Corona-Warn-App
+//
+// SAP SE and all other contributors
+// copyright owners license this file to you under the Apache
+// License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import XCTest
+import ExposureNotification
+@testable import ENA
+
+class ExposureManagerTests: XCTestCase {
+
+	func test_GIVEN_RunningDetection_When_StartNewDetection_Then_SameProgressReturned() {
+		let managerSpy = ManagerSpy()
+		let exposureManager = ENAExposureManager(manager: managerSpy)
+
+		let progress1 = exposureManager.detectExposures(configuration: ENExposureConfiguration(), diagnosisKeyURLs: []) { _, _ in
+		}
+
+		let progress2 = exposureManager.detectExposures(configuration: ENExposureConfiguration(), diagnosisKeyURLs: []) { _, _ in
+		}
+
+		XCTAssertEqual(progress1, progress2)
+	}
+
+	func test_GIVEN_CanceledDetection_When_StartNewDetection_Then_NewProgressReturned() {
+		let managerSpy = ManagerSpy()
+		let exposureManager = ENAExposureManager(manager: managerSpy)
+
+		let progress1 = exposureManager.detectExposures(configuration: ENExposureConfiguration(), diagnosisKeyURLs: []) { _, _ in
+		}
+
+		progress1.cancel()
+
+		let progress2 = exposureManager.detectExposures(configuration: ENExposureConfiguration(), diagnosisKeyURLs: []) { _, _ in
+		}
+
+		XCTAssertNotEqual(progress1, progress2)
+	}
+
+	func test_GIVEN_FinishedDetection_When_StartNewDetection_Then_NewProgressReturned() {
+		let managerSpy = ManagerSpy()
+		let exposureManager = ENAExposureManager(manager: managerSpy)
+
+		let progress1 = exposureManager.detectExposures(configuration: ENExposureConfiguration(), diagnosisKeyURLs: []) { _, _ in
+		}
+
+		// Simulate isFinished state.
+		progress1.totalUnitCount = 1
+		progress1.completedUnitCount = 1
+
+		let progress2 = exposureManager.detectExposures(configuration: ENExposureConfiguration(), diagnosisKeyURLs: []) { _, _ in
+		}
+
+		XCTAssertNotEqual(progress1, progress2)
+	}
+
+	func test_GIVEN_RunningDetection_When_StartNewDetection_Then_DetectExposuresNotCalled() {
+		let managerSpy = ManagerSpy()
+		let exposureManager = ENAExposureManager(manager: managerSpy)
+
+		_ = exposureManager.detectExposures(configuration: ENExposureConfiguration(), diagnosisKeyURLs: []) { _, _ in
+		}
+
+		_ = exposureManager.detectExposures(configuration: ENExposureConfiguration(), diagnosisKeyURLs: []) { _, _ in
+		}
+
+		XCTAssertEqual(managerSpy.numberOfDetectionCalls, 1)
+	}
+}
+
+private final class ManagerSpy: NSObject, Manager {
```

Comment:
Is this a new fancy name for a mock manager? 🕵🏼‍♂️😎

Reply:
It's just a more precise naming for this Double. I first read about it from Martin Fowler: https://martinfowler.com/articles/mocksArentStubs.html

Here is another article from objc.io: 
https://www.objc.io/issues/15-testing/mocking-stubbing/